### ISSUE TITLE
Customizable character counter colour for Textbox

### ISSUE THREAD

**Issue by IrynaD26** [Sage/carbon#7267]
### Description

As a developer using Carbon, I want to be able to change the character counter colour,
This will help us avoid accessibility issues related to the insufficient contrast because of the character counter colour.

### Suggested solution

Adding a prop to Textbox for changing character counter colour.

### Confidentiality

- [x] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
@ljemmo @harpalsingh are you happy to support this feature?
